storage/storage_dest.go

```diff
@@ -930,7 +930,7 @@ func (s *storageImageDestination) createNewLayer(index int, layerDigest digest.D
 
 		flags := make(map[string]interface{})
```

Comment:
Why was `map[string]interface{}` chosen here? It came in 15ac716f1690cb04f3694e7b295e4830217e8c62 but I don't understand why it couldn't be a structure.

Reply:
See how it is used in that commit; this goes all the way back to 7426cceb6f .

As for why this does not record specific fields … \*shrug\* it’s an escape hatch so that out-of-c/storage users can store their private data without _properly_ maintaining a c/storage type extension or without making that data public to all c/storage users; and that’s basically what’s happening here.